include/seastar/net/socket_defs.hh

```diff
@@ -123,6 +133,14 @@ struct ipv6_addr {
     }
 };
 
+struct ud_addr {
+    std::filesystem::path sfile_;
+
+    explicit ud_addr(const std::string&);
+    explicit ud_addr(std::string_view fn) { sfile_ = fn; }
+    explicit ud_addr(const std::filesystem::path& fn) { sfile_ = fn; }
```

Comment:
please use member initializer list to avoid creating an instance and then throw it away when assigning to it.

Reply:
fixed - but note that I checked the resulting assembly both ways - and the optimizer produces the same code.

Reply:
thanks for checking the assembly. but i'd rely on the behavior defined by standard as long as it's defined.